source/incqueryserver-jupyter/iqs_jupyter/core_extensions.py

```diff
@@ -386,8 +390,29 @@ def __init__(
         else:
             fields = [self.address_field]
         
-        self.box=widgets.HBox([widgets.Label(value=labelText), widgets.VBox(fields)])
-        if auto_display: self.display()
+        self.box = widgets.HBox([widgets.Label(value=labelText), widgets.VBox(fields)])
+        if auto_display:
+            self.display()
+
+        if login_button:
+            btn_login = interact.options(manual=True, manual_name="Login")
```

Comment:
From a UX perspective, it is weird that you may or may not click a Login button, but then have to execute a `connect()` call regardless. Perhaps labeling the button "Test Connection" would make the user better understand what the consequences are. (If we decide to change the label, we should also change variable and parameter names accordingly.) What do you think?

Reply:
My intention with the Login button was to replace the need for a `connect()` call. That way the button should be mandatory (instead of an optional parameter), and should be clicked. You are right, that if this button is optional, the "Test Connection" may be a better representation of the feature.

Although another reason for this enhancement was, to make it easier to change connections between IQS instances, the interaction with the button makes the situation harder to implement. If the widget had an `IQSClient` property, that we can reference at all times, we could solve this problem.

Then we could reference the running IQS connection like this:
```python
connector = iqs_jupyter.IQSConnectorWidget()
# [ ... ]
# instead of calling IQSConnectorWidget.connect()
connector.iqs.repository.update_repository()
```

Reply:
@Sealdolphin right now the `connect()` function is pretty much the same as this property. We can change it so that it is synonymous with the `__get__` of this property.

As for the button being mandatory: we should at least allow the parametrization to opt out, so that no manual login is required (i.e. the login button is automatically pressed to connect with the field values provided as parameters); this would be important for automated / headless execution of the notebook.

BTW (does not have to be within the scope of this issue) if we can make the property a traitlet as well, that it could be databound to inputs of other widgets in the future, so that they can update in an event-driven way if the IQS is (re)connected.

Reply:
> we should at least allow the parametrization to opt out, so that no manual login is required

I agree. Absolutely.

> We can change it so that it is synonymous with the __get__ of this property

Then I think this is where I want to go.

> if we can make the property a traitlet as well

That sounds really good.